src/http/register-websocket/connection.js

```diff
@@ -1,54 +1,80 @@
+let makeRequestId = require('../../lib/request-id')
 let invoke = require('../invoke-ws')
 let pool = require('./pool')
 let noop = err => err ? console.log(err) : ''
 
-module.exports = function connection ({ cwd, inventory, update }, connectionId, ws) {
+module.exports = function connection ({ cwd, inventory, update, connectedAt, domainName, stage }, connectionId, ws) {
   let { get } = inventory
-
   // Save this for send to use
   pool.register(connectionId, ws)
 
+  let makeRequestContext = (extra) => ({
+    messageId: makeRequestId(),
+    messageDirection: 'IN',
+    connectedAt,
+    requestTimeEpoch: Date.now(),
+    requestId: makeRequestId(),
+    connectionId,
+    domainName,
+    stage,
+    ...extra
+  })
+
```

Comment:
Hoping we could make sure that all logic pertaining to the construction of request objects to live in `src/http/invoke-$TYPE/**/_req-fmt.js` – will dry up the shape, and I'm not seeing anything here that necessitates this logic lives in both `connection.js` and `upgrade.js`.